components/sessionModals/SessionTimeTracker.js

```diff
@@ -0,0 +1,67 @@
+import Modal from 'react-modal'
+import { useState, useEffect, useCallback } from 'react'
+import CountDown from './CountDown'
+import Router from 'next/router'
+import { useIdleTimer } from 'react-idle-timer'
+
+const SessionTimeTracker = ({ popupStaySignedIn, aaPrefix }) => {
+  const [demoModalBody, setDemoModalBody] = useState(null)
+
+  const handleOnIdle = () => {
+    Router.push('/auth/logout')
+  }
+
+  const { isPrompted, reset, getRemainingTime } = useIdleTimer({
+    onIdle: handleOnIdle,
+    promptBeforeIdle: 0.5 * 60 * 1000, // 30 seconds
+    timeout: 1 * 60 * 1000, // 1 minute
```

Comment:
This amount of time could be bad UX. I think the decision should ultimately be up to Farwa but I'd have the timeout set to 10 minutes and the prompt before logging the user out to be 5 minutes.